akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -229,6 +254,39 @@ private[akka] object StreamLayout {
     }
 
     /**
+     * Creates a new Module which is `this` Module composed with `that` Module.
+     *
+     * The difference to compose(that) is that this version completely ignores the materialized value
+     * computation of `that` while the normal version executes the computation and discards its result.
+     * This means that this version must not be used for user-provided `that` modules, it can only be
+     * used where we know that there is no meaningful computation to be done (like for
+     * MaterializedValueSource).
```

Comment:
maybe add something like: because in user facing DSLs we need to call mapMaterialized for possible side-effects